### PR TITLE
Added the functionality to print DiracDelta using δ(x)

### PR DESCRIPTION
**Pull request by sampadsaha5**
I have added the functionality, but I am not pretty sure about it. Locally, the tests are running fine. 

The desired Functionality is: 

``` py
>>> DiracDelta(x)
δ(x)

>>> DiracDelta(x-4)
δ(x - 4)
```

@moorepants @leosartaj

### CODE REVIEW

File: sympy/printing/pretty/pretty.py [hunk 1]
```diff
@@ -1149,6 +1149,15 @@ def _print_gamma(self, e):
         else:
             return self._print_Function(e)
 
+    def _print_DiracDelta(self, e):
+        if self._use_unicode:
+            pform = self._print(e.args[0])
```
Comment (leosartaj) [inline sympy/printing/pretty/pretty.py]: Why just args[0] and not the second arg?

Comment (sampadsaha5) [inline sympy/printing/pretty/pretty.py]: Oops.. it got out of my mind.

**Comment by leosartaj** [comment]
You can also enable pretty printing in latex. See [this](https://github.com/sympy/sympy/blob/master/sympy/printing/latex.py).

**Comment by sampadsaha5** [comment]
Ok, I will also try to enable pretty printing in latex.

**Comment by sampadsaha5** [comment]
@leosartaj  pretty printing in latex for `DiracDelta` is already there. 
Check [this](https://github.com/sympy/sympy/blob/master/sympy/printing/latex.py#L1430).

**Comment by leosartaj** [comment]
This is how [latex](https://gist.github.com/leosartaj/2267f0bb8fbd198de71312f9c3ffedb8) printing looks like for ``DiracDelta(x, 1)``.

File: sympy/printing/pretty/tests/test_pretty.py [hunk 2]
```diff
@@ -4377,6 +4377,11 @@ def test_gammas():
     assert xpretty(gamma(x), use_unicode=True) == u'Γ(x)'
 
 
+def test_deltas():
+    assert xpretty(DiracDelta(x), use_unicode=True) == u'δ(x)'
+    assert xpretty(DiracDelta(x, 1), use_unicode=True) == u'δ(x, 1)'
```
Comment (moorepants) [inline sympy/printing/pretty/tests/test_pretty.py]: Should this print as:

```
d δ(x)
--------
dx
```

And what should something like `DiracDelta(x - 1)` print as?

Comment (leosartaj) [inline sympy/printing/pretty/tests/test_pretty.py]: IMHO ``DiracDelta(x - 1)`` should print as ``δ(x - 1)``. I like the way derivatives are printed in latex. See [this](https://gist.github.com/leosartaj/2267f0bb8fbd198de71312f9c3ffedb8).

**Comment by leosartaj** [comment]
Have you given any thought on how to print ``DiracDelta`` derivatives?

**Comment by sampadsaha5** [comment]
```
dδ(x)
-----
dx
```
Seems good to me.

**Comment by leosartaj** [comment]
We should try to keep latex printing and pretty printing similar.

**Comment by moorepants** [comment]
Is the LaTeX printing common for derivatives? I've never seen that notation.

**Comment by sampadsaha5** [comment]
To keep them same, `δ(x, 1)` looks good to me.

**Comment by moorepants** [comment]
Should DiracDelta only accept two types of arguments:

1. `x` (i.e. a symbol)
2. `x - a` (i.e. the shifted x)

If so, wouldn't this API be more appropriate:

`DiracDelta(x, shift=0, derivative=0)`

with only things like the following allowed:

```pycon
>>> x, y = symbols('x, y')
>>> DiracDelta(x, shift=2)
δ(x - 2)
>>> DiracDelta(y)
δ(y)
>>> DiracDelta(x, derivative=1)
d δ(x)
------
dx
```

Should we support arbitrary expressions being passed into DiracDelta? If not, that simplifies a lot of things. We've been discussing issues with complex x or a complex expression being passed in, But maybe we are making this too complicated.

**Comment by moorepants** [comment]
I see in this Wolfram article (http://mathworld.wolfram.com/DeltaFunction.html) that delta accepts more arbitrary inputs.

**Comment by moorepants** [comment]
That article also has notation for the derivative. I'm guessing whoever wrote our DiracDelta class followed that article closely, so maybe we should stick to that.

**Comment by leosartaj** [comment]
> Should DiracDelta only accept two types of arguments:

> x (i.e. a symbol)
x - a (i.e. the shifted x)
If so, wouldn't this API be more appropriate:

> DiracDelta(x, shift=0, derivative=0)

This looks similar to what we discussed. I like setting ``derivative=0``. I am not sure if we want ``shift=0`` here though. ``DiracDelta(x - 5)`` seems more intuitive to me from a user's perspective.
 
> Should we support arbitrary expressions being passed into DiracDelta? If not, that simplifies a lot of things. We've been discussing issues with complex x or a complex expression being passed in, But maybe we are making this too complicated.

See if this is [works](https://gitter.im/sympy/sympy?at=5744add5fce033da4befa116).

**Comment by leosartaj** [comment]
> That article also has notation for the derivative. I'm guessing whoever wrote our DiracDelta class followed that article closely, so maybe we should stick to that.

+1

**Comment by sampadsaha5** [comment]
+1 
for the idea to keep `derivative=0` but if we use `shift =0` then how will we represent                                               `DiracDelta(x**2 - x - 1)`

**Comment by moorepants** [comment]
The shift=0 was a bad idea. I didn't realize there was a need to provide arbitrary functions to dirac delta. I see how passing in polynomials can allow for multiple spikes, etc.

**Comment by sampadsaha5** [comment]
We can implement `derivative = 0` in a different PR. I think we should open an issue so that everyone in the community can discuss their views over this API change.
How does it sound?

**Comment by sampadsaha5** [comment]
Does this PR need any modifications?
Any suggestions are welcome.

**Comment by leosartaj** [comment]
I feel you should change pretty printing of derivatives similar to latex printing.

**Comment by sampadsaha5** [comment]
Do you mean like [this](https://gist.github.com/leosartaj/2267f0bb8fbd198de71312f9c3ffedb8).

**Comment by moorepants** [comment]
Use the top one as that matches the wolfram page and the current latex.

**Comment by sampadsaha5** [comment]
I will try to do that. Can you please give a reference from where I can get to know how this `(1)` can be printed like the way we wanted?

**Comment by moorepants** [comment]
Look in the pretty printing module and see if you can find some unicode superscripts. Or look up unicode super scripts in google.

**Comment by sampadsaha5** [comment]
Thanks, Jason. I will try to look up for unicode superscripts at google

**Comment by sampadsaha5** [comment]
I tried doing this and it gave a result which looks almost similar to the one which we wanted(it may be wrong I am not sure):
```py
In [54]: from sympy.printing.pretty.stringpict import stringPict

In [55]: from sympy.printing.pretty.pretty_symbology import greek_unicode

In [56]: a =stringPict("1").parens()[0]

In [57]: print(stringPict(greek_unicode['delta'] + "   (x)").above(a)[0])
  (1)  
δ   (x)


```

**Comment by moorepants** [comment]
That looks good.

**Comment by sampadsaha5** [comment]
But if we do that way, this doesn't look nice:
```py
In [7]: print(stringPict(greek_unicode['delta'] + "  (x**2 + 1)").above(a)[0])
     (1)     
δ  (x**2 + 1)

```

**Comment by moorepants** [comment]
Any idea why the superscript floats so far to the right?

**Comment by sampadsaha5** [comment]
I think `above()` adjusts the upper expression according to the length of the lower expression.
It sets the upper in the center of the lower.

**Comment by moorepants** [comment]
above probably isn't the right thing to use. Unicode supports superscripts, not sure about ones in parens.

**Comment by moorepants** [comment]
You want something like this:

```
In [31]: print(prettyForm(greek_unicode['delta'])**prettyForm('(1)'))
 (1)
δ   
```

**Comment by sampadsaha5** [comment]
Thanks Jason. Yah that is what I was looking for..
Let figure out how to add this in the pprint module.

**Comment by sampadsaha5** [comment]
Please help me with this:- 
```
In [10]: print(prettyForm(greek_unicode['delta']))
---------------------------------------------------------------------------
UnicodeEncodeError                        Traceback (most recent call last)
<ipython-input-10-eb959ac46596> in <module>()
----> 1 print(prettyForm(greek_unicode['delta']))

UnicodeEncodeError: 'ascii' codec can't encode character u'\u03b4' in position 0: ordinal not in range(128)

```

**Comment by moorepants** [comment]
The above works for Python 3.5 but not 2.7. Here is a clue for a fix:
```
In [2]: from sympy.printing.pretty.stringpict import prettyForm

In [3]: from sympy.printing.pretty.pretty_symbology import greek_unicode

In [4]: print(prettyForm(greek_unicode['delta']))
---------------------------------------------------------------------------
UnicodeEncodeError                        Traceback (most recent call last)
<ipython-input-4-eb959ac46596> in <module>()
----> 1 print(prettyForm(greek_unicode['delta']))

UnicodeEncodeError: 'ascii' codec can't encode character u'\u03b4' in position 0: ordinal not in range(128)

In [6]: greek_unicode['delta']
Out[6]: u'\u03b4'

In [7]: prettyForm(greek_unicode['delta'])
Out[7]: prettyForm(u'\u03b4',0,0)

In [8]: a = greek_unicode['delta']

In [9]: prettyForm(a.encode('utf-8'))
Out[9]: prettyForm('\xce\xb4',0,0)

In [10]: print(prettyForm(a.encode('utf-8')))
δ
```

**Comment by sampadsaha5** [comment]
I have implemented this: 
```py
In [4]: pprint(xpretty(DiracDelta(x-1, 10), use_unicode=True))
 (10)        
δ     (x - 1)

In [5]: pprint(xpretty(DiracDelta(x**2-1, 10), use_unicode=True))
 (10) ⎛ 2    ⎞
δ     ⎝x  - 1⎠

```
Is this fine?

**Comment by moorepants** [comment]
That looks great.

**Comment by sampadsaha5** [comment]
Thanks @moorepants.

**Comment by leosartaj** [comment]
Good Work!

**Comment by sampadsaha5** [comment]
Thanks Sartaj. :smile:

**Comment by asmeurer** [comment]
This has led to a regression https://github.com/sympy/sympy/issues/14101

**Pull request merged**